QuEST/include/QuEST.h

```diff
@@ -2447,6 +2447,25 @@ void applyTwoQubitKrausMap(Qureg qureg, int target1, int target2, ComplexMatrix4
  */
 qreal calcHilbertSchmidtDistance(Qureg a, Qureg b);
 
+/** Computes the Hilbert-Schmidt scalar product between two density matrices \p a and \p b, 
+ * equialent to the Frobenius inner product of matrices.
+ * That is, we define the Hilbert-Schmidt scalar product
+ * \f[
+    ((a, b))_HS = (( a, b ))_F = \text{Tr}[ a^\dagger b ] 
+ * \f]
+ * This is equivalent to the sum of products of matrix elemets, i.e.
+ * \f[
+    ((a, b))_HS = \sum\limits_i \sum\limits_j  conj(a_{ij}) b_{ij}
+ * \f]
+ *
+ * @param[in] a a density matrix
+ * @param[in] b an equally-sized density matrix
+ * @throws exitWithError
+ *      if either \p a or \p b are not density matrices,
+ *      or if \p a and \p have mismatching dimensions.
+ */
+qreal calcHilbertSchmidtScalarProduct(Qureg a, Qureg b);
```

Comment:
If the Hilbert Schmidt scalar product is a natural choice of inner product for density matrices, i.e. one analogous to `<a|b>`, then perhaps it's better to "overload" `calcInnerProduct`, and i.e. make this method `densmatr_calcInnerProduct`.

That way, user's code for processing statevectors is easily adapted to density matrices to e.g. introduce noise

Reply:
fixed this in commits 7af7983c750036da33a3074eb7462164c66e5368 and 8de0d6f2e62f1a2f150a7b26ae5ad13e28b6b53b

Reply:
(note I think the above response was meant for the previous QuEST.h changes; those commits do not address this review point)

Reply:
Sorry, that comment was ment to be for the above QuEST.h changes.

I updated the calcInnerProduct(...) function such that it now accepts density matrices and returns their Hilbert-Schmidt scalar product. See commits 1ea8ceb28a4a7b3a8025ce92b9129b894100911 
6c9f711d396ad704ef0aaaed8e9d04b80b19685d and 691f66020ba94a8149e7abc79619de06b4fae1b0